book/12-d-csound-in-ios.md

```diff
@@ -1008,19 +1008,16 @@ The process demonstrated in the code examples below can be described as:
 }
 ```
 
-Note that in Swift, we have to create a `CVaListPointer` (equivalent to a
-`va_list *` in C) for use with the `vsnprintf()` function:
+
 
 ```swift
 // Swift
 func messageCallback(_ infoObj: NSValue) {
     var info = Message()
     infoObj.getValue(&info)
     let message = UnsafeMutablePointer<Int8>.allocate(capacity: 1024)
-    let va_ptr: CVaListPointer = CVaListPointer(
-      _fromUnsafeMutablePointer: &(info.valist)
-    )
-    vsnprintf(message, 1024, info.format, va_ptr)
+
+    vsnprintf(message, 1024, info.format, info.valist)
     let messageStr = String(cString: message)
     print(messageStr)
```

Comment:
Also, the memory for buffer should be deallocated in the end to prevent excessive memory use:
`            buffer?.deallocate()
`